test/parallel/test-tls-timeout-server-2.js

```diff
@@ -1,32 +1,32 @@
 'use strict';
-var common = require('../common');
-var assert = require('assert');
+const common = require('../common');
+const assert = require('assert');
 
 if (!common.hasCrypto) {
   common.skip('missing crypto');
   return;
 }
-var tls = require('tls');
+const tls = require('tls');
 
-var fs = require('fs');
+const fs = require('fs');
 
-var options = {
+const options = {
   key: fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
   cert: fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem')
 };
 
-var server = tls.createServer(options, function(cleartext) {
-  var s = cleartext.setTimeout(50, function() {
+const server = tls.createServer(options, common.mustCall(function(cleartext) {
+  const s = cleartext.setTimeout(50, common.mustCall(function() {
```

Comment:
Don't we want to ensure that `cleartext.setTimeout` actually calls its callback? I get that we don't want to use `common.mustCall` when there are no assertions, so I'm thinking we want to actually add an assertion that the function is called (e.g., by using `assert.ok(true)` within the function body or whatever is most idiomatic).

I'm not entirely sure what the intention behind this test file is; is it really just to check the type of `tls.createServer`'s argument? Given the name of the test file, I figured it was that the timeout actually gets called.

I took a look at the other test file for [`test-tls-timeout-server`](https://github.com/nodejs/node/blob/8264a2225628cbc32116f83a079f6024786988ab/test/parallel/test-tls-timeout-server.js) and it seems to just be testing the `tlsClientError` event.

Incidentally, on [line 21](https://github.com/nodejs/node/blob/8264a2225628cbc32116f83a079f6024786988ab/test/parallel/test-tls-timeout-server.js#L21) it does something similar to what I was doing in this file. Depending on the outcome of our conversation here, it seems as though that file should be changed as well?